community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -0,0 +1,452 @@
+                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                {
+                    throw new UnsupportedOperationException( "Add support for arrays" );
+                }
+
+                private boolean valueIsInlined( long firstBlock )
+                {
+                    // [][][][i][ssss,tttt][kkkk,kkkk][kkkk,kkkk][kkkk,kkkk]
+                    return (firstBlock & 0x100000000L) > 0;
+                }
+            },
+    TEMPORAL_LOCAL_DATE_TIME( 3, "LocalDateTime" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    long nanoOfSecond = valueBlocks[offset] >>> 32;
+                    long epochSecond = valueBlocks[1 + offset];
+                    return LocalDateTimeValue.localDateTime( epochSecond, nanoOfSecond );
+                }
+
+                @Override
+                public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
+                {
+                    return 2;
+                }
+
+                @Override
+                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                {
+                    throw new UnsupportedOperationException( "Add support for arrays" );
+                }
+            },
+    TEMPORAL_TIME( 4, "Time" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    int secondOffset = (int) (valueBlocks[offset] >>> 32);
+                    long nanoOfDay = valueBlocks[1 + offset];
+                    return TimeValue.time( nanoOfDay, ZoneOffset.ofTotalSeconds( secondOffset ) );
+                }
+
+                @Override
+                public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
+                {
+                    return 2;
+                }
+
+                @Override
+                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                {
+                    throw new UnsupportedOperationException( "Add support for arrays" );
+                }
+            },
+    TEMPORAL_DATE_TIME( 5, "DateTime" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    if ( storingZoneOffset( valueBlocks[0] ) )
+                    {
+                        int nanoOfSecond = (int) (valueBlocks[offset] >>> 33);
+                        long epochSecond = valueBlocks[1 + offset];
+                        int secondOffset = (int) valueBlocks[2 + offset];
+                        return DateTimeValue.datetime( epochSecond, nanoOfSecond, ZoneOffset.ofTotalSeconds( secondOffset ) );
+                    }
+                    else
+                    {
+                        throw new UnsupportedOperationException( "..." );
```

Comment:
Likely to happen? Should there be a better description of this error?

Reply:
Sorry, this is when we encounter a DateTime with ZoneId and the else block will then be implemented as soon as we support that format.